sys/shell/commands/sc_gnrc_icmpv6_echo.c

```diff
@@ -339,7 +339,8 @@ static void _print_reply(_ping_data_t *data, gnrc_pktsnip_t *icmpv6,
 
         /* not our ping */
         if ((byteorder_ntohs(icmpv6_hdr->id) != data->id) ||
-            !ipv6_addr_equal(from, &data->host)) {
+            !(ipv6_addr_is_multicast(&data->host) &&
```

Comment:
```suggestion
            !(ipv6_addr_is_multicast(&data->host) ||
```

But with this, broadcast ping is working again :)

Reply:
Or to avoid these kind of errors once and for all:

```C
        /* not our ping */
        if (byteorder_ntohs(icmpv6_hdr->id) != data->id) {
            return;
        }

        if (!ipv6_addr_is_multicast(&data->host) &&
            !ipv6_addr_equal(from, &data->host)) {
            return;
        }
```

Reply:
Arghs ... foiled by deMorgans law again -.-

Reply:
Fixed